### PR TITLE
perf(core): precompile hex color regex pattern at module level

### PR DESCRIPTION
**Pull request by Rudra-Tiwari-codes**
Moves hex color validation regex from inside `_render_mermaid_using_api()` to module-level constant `_HEX_COLOR_PATTERN`. This avoids recompiling the regex on every function call, improving performance when rendering multiple Mermaid graphs.

**Testing:**
- `make lint` passes
- `make test` passes

### CODE REVIEW

File: libs/core/langchain_core/runnables/graph_mermaid.py [hunk 1]
```diff
@@ -40,6 +40,9 @@
 
 MARKDOWN_SPECIAL_CHARS = "*_`"
 
+# Precompile regex pattern for hex color validation
```
Comment (mdrxy) [inline libs/core/langchain_core/runnables/graph_mermaid.py]:
```suggestion
```

**Review by mdrxy — APPROVED** [review]
Thank you!

**Comment by fransafu** [comment]
@mdrxy 

**Warning:** This author is forking multiple ML projects such as google-deepmind/alphafold, ml-explore/mlx, openai/CLIP, pytorch/pytorch, tensorflow/tensorflow, anthropics/claude-code, vllm-project/vllm, and others, adding minimal "contributions" (often for tests or miscellaneous changes) without proper validation. A review of their commits shows mostly local implementations of TODOs copied from existing projects, with little to no substantive review or testing.

So far, this author has forked **41 repositories** following the same pattern. Be careful when accepting this PR. It’s also concerning how this author is able to submit PRs across four repositories in the same day, each requiring large context, which strongly suggests a highly automated workflow.

**Comment by Rudra-Tiwari-codes** [comment]
Thank you for the feedback regarding my recent activity. I am a student and I have been using these smaller tasks as a way to familiarize myself with the architecture of various codebases. I was not aware that submitting multiple minor pull requests was considered disruptive to the maintainer workflow or seen as contribution padding. I appreciate the correction and will pay much closer attention to the impact of my work moving forward. I am closing this pull request now to focus on delivering more substantive technical contributions that provide genuine value to the community.
